implementation/src/main/java/io/smallrye/mutiny/Uni.java

```diff
@@ -371,6 +372,22 @@ static UniCombine combine() {
         return onItem().produceUni(nonNull(mapper, "mapper"));
     }
 
+    /**
+     * Produces a new {@link Uni} invoking the given callback when the {@code item} event is fired. Note that the
+     * item can be {@code null}.
+     * <p>
+     * While this method allows implementing side-effects <em>sneakily</em>, it is recommended to not do so, and only use
+     * this method to execute side-effect free synchronous logic, like tracing.
+     *
+     * This method is a shortcut on {@code onItem().invoke(consumer)}.
+     *
+     * @param callback the callback, must not be {@code null}
+     * @return the new {@link Uni}
+     */
+    default Uni<T> sneak(Consumer<? super T> callback) {
```

Comment:
"`sneak()`", seriously?

So my Hibernate code is going to be full of `.sneak( item -> session.remove(item) )`?

You're basically signaling that what is bog-standard Hibernate Reactive code is a misuse of your API.

Reply:
It was `peek`, but after discussing with @kenfinnigan we replaced it with `sneak`.

Reply:
OK, but we were already discussing this in an issue.

Reply:
I liked `use()` but @cescoffier pointed out it implies the item isn't passed along the pipeline, whereas it is

Reply:
I can remove the javadoc sentence. BTW session.remove is synchronous?

Reply:
@kenfinnigan how does it imply that? Using a hammer doesn't destroy it. I can use a hammer without having to throw it away afterward.

Reply:
@cescoffier 

> BTW session.remove is synchronous?

Sorry, bad example, better examples would be:

    .sneak( item -> item.setThing(thing) )

Or:

    .sneak( session -> session.close() )

This isn't Haskell, folks.

Reply:
That was more my understanding of `use`.

```
multi
 .map(it -> ...)
 .use(x -> doSomethingWith(x))
 .map( /* not clear which item would get here - is it x, is it something else? */)
 .onFailure().recoverWithCompletion();
```

Reply:
Well there are two operations, both very important and useful:

- `apply()` accepts a `Function<T,S>` and gives you back a `Uni<S>`
- `use()` or `invoke()` or `accept()` or whatever takes a `Consumer<T>` and gives you back a `Uni<T>`

Now, for some dumb reason `CompletionStage` has a broken version of `accept()` which gives you back a `CompletionStage<Void>`, forcing you to have to use `apply()` when you would really prefer to use `accept()`. But if you get the return type correct then both are useful and important.

Reply:
(And then of course you need the lifted versions `applyUni()` and `useUni()` or whatever.)

Reply:
Actually, I realized the javadoc comment is a missed of side-effect. I want to prevent concurrent access to shared resources/data.